.github/workflows/docker.yml

```diff
@@ -0,0 +1,61 @@
+on:
+  workflow_dispatch:
+    inputs:
+      image-prefix:
+        description: "gets suffixed with 'base' and 'sdk' to create actual image name"
+        default: ghcr.io/viamrobotics/cpp-
+      dockerfile:
+        default: Dockerfile.debian.bullseye
+      tag:
+        default: bullseye-amd64
+      build-base:
+        description: "whether to build the base image. the base images change less often and may not be necessary to rebuild."
+        type: boolean
+        default: false
+      build-sdk:
+        description: "whether to build the SDK image. if this is true and no corresponding base image exists, the job will fail."
+        type: boolean
+        default: false
+      push:
+        description: "whether to push the images after building them"
+        type: boolean
+        default: false
+
+jobs:
+  build-container:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - uses: actions/checkout@v4
+
+    # build base (if inputs.build-base)
+    - uses: docker/metadata-action@v5
+      id: base-meta
+      if: inputs.build-base
+      with:
+        images: ${{ inputs.image-prefix }}base
+    - uses: docker/build-push-action@v5
+      if: inputs.build-base
+      with:
+        push: ${{ inputs.push }}
+        tags: "${{ inputs.image-prefix }}base:${{ inputs.tag }}"
+        file: etc/docker/base-images/${{ inputs.dockerfile }}
+        labels: ${{ steps.base-meta.output.labels }}
+
+    # build sdk (if inputs.build-sdk)
```

Comment:
Can you explain the purpose of separating `build-base` and `build-sdk` into two separate steps? are there times where only one would be built?

Reply:
I did this because the instructions in the readme were already two-step

but in general the purpose of 'base images' is to save time on the constant part of the build

here specifically, could imagine us rebuilding the base images weekly or monthly, and rebuilding the sdk image on every release or on main branch update